app/components/Project/Project.js

```diff
@@ -305,6 +305,35 @@ class Project extends Component<Props> {
     }
   };
 
+  checklistUpsertNoteHandler = (checklistItem, text, note) => {
+    if (this.unchangedNote(note,text)) {
+      return;
+    }
+
+    const currentProject = { ...this.props.project };
+    const action = { type: '', title: '', description: '', details: null };
+    this.upsertNoteHandler(
+      checklistItem,
+      EntityType.CHECKLIST,
+      checklistItem.name,
+      action,
+      text,
+      note
+    );
+    const updatedChecklist = this.props.checklistResponse.checklist.map(x => {
```

Comment:
I understand what this code is doing, but am curious why this extra call to update the checklist is needed each time a note is added/updated/deleted? Maybe some comments in the code around this portion would be helpful.

Reply:
yeah, i knew this might cause some confusion!
actually the upsert and delete note-handlers only return us the updated entity (checklistItem in this case), but that doesn't save the updates, to save them we need to update whatever main object they are attached to, checklist is on it's own, unlike the asset or person which are attached to project object, so we update the checklist that's associated with the current project. so, `onChecklistUpdated` as opposed to `onUpdated` which is for project. all this is in reference to the other update or delete handlers (personUpsertNoteHandler, projectUpsertNoteHandler, assetUpsertNoteHandler).

Reply:
Thank you!!!